src/main/scala/org/ergoplatform/ErgoLikeContext.scala

```diff
@@ -25,6 +30,57 @@ class ErgoLikeContext(val currentHeight: Height,
 
   def withTransaction(newSpendingTransaction: ErgoLikeTransactionTemplate[_ <: UnsignedInput]): ErgoLikeContext =
     ErgoLikeContext(currentHeight, lastBlockUtxoRoot, boxesToSpend, newSpendingTransaction, self, extension)
+
+  val Cols: ColOverArrayBuilder = new ColOverArrayBuilder
+
+  def collection[T:ClassTag](items: T*) = Cols.fromArray(items.toArray)
+
+  val noBytes = collection[Byte]()
+  val noInputs = Array[Box]()
+  val noOutputs = Array[Box]()
+  val emptyAvlTree = new TestAvlTree(noBytes, 0, None, None, None)
+
+
+  def regs(m: Map[Byte, Any]): Col[AnyValue] = {
+    val res = new Array[AnyValue](10)
+    for ((id, v) <- m) {
+      assert(res(id) == null, s"register $id is defined more then once")
+      res(id) = new TestValue(v)
+    }
+    Cols.fromArray(res)
+  }
+
+  def contextVars(m: Map[Byte, Any]): Col[AnyValue] = {
+    val maxKey = if (m.keys.isEmpty) 0 else m.keys.max
+    val res = new Array[AnyValue](maxKey)
+    for ((id, v) <- m) {
+      val i = id - 1
+      assert(res(i) == null, s"register $id is defined more then once")
+      res(i) = new TestValue(v)
+    }
+    Cols.fromArray(res)
+  }
+
+  override def toSigmaContext(IR: Evaluation): sigma.Context = {
+    implicit val IRForBox = IR
+    val inputs = boxesToSpend.toArray.map(_.toTestBox(true))
```

Comment:
@greenhat `isCost == true` only for context which is prepared for evaluation of cost function.